src/aws-cpp-sdk-core/include/smithy/client/features/ChecksumInterceptor.h

```diff
@@ -233,6 +172,94 @@ class ChecksumInterceptor : public smithy::interceptor::Interceptor {
     }
   }
 
+  void handleProvidedChecksum(const Aws::AmazonWebServiceRequest& request, std::shared_ptr<Aws::Http::HttpRequest> httpRequest,
+                              const Aws::String& algorithm, const Aws::String& checksumType, const Aws::String& checksumValue) {
+    if (request.IsStreaming()) {
+      addChecksumFeatureForChecksumName(algorithm, request);
+      if (httpRequest->GetRequestHash().second == nullptr) {
+        auto hash = Aws::MakeShared<PrecalculatedHash>(AWS_SMITHY_CLIENT_CHECKSUM, checksumValue);
+        httpRequest->SetRequestHash(algorithm, hash);
+      }
+    } else {
+      httpRequest->SetHeaderValue(checksumType, checksumValue);
+    }
+  }
+
+  void calculateAndSetChecksum(const Aws::AmazonWebServiceRequest& request, std::shared_ptr<Aws::Http::HttpRequest> httpRequest,
+                               const Aws::String& algorithm, const Aws::String& checksumType) {
+    static const Aws::Array<std::pair<const char*, ChecksumHandler>, 5> algorithmMap = {{
```

Comment:
should declare it as static in the compilation unit i.e. just in the `.cpp`. no reason to have it in the function.